controllers/storagecluster/storagecluster_controller.go

```diff
@@ -166,13 +166,12 @@ func (r *StorageClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 	}
 
-	onboardingSecretPredicates := builder.WithPredicates(
-		predicate.NewPredicateFuncs(
-			func(client client.Object) bool {
-				return client.GetName() == onboardingTicketPublicKeySecretName
-			},
-		),
-	)
+	secretPredicate := predicate.Funcs{
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			// Evaluates to false if the object has been confirmed deleted.
+			return !e.DeleteStateUnknown
```

Comment:
I do not understand what this field is actually used for even after going through the docs and you don't want to invoke reconcile for every secret deletion (well, nothing usually deletes secrets again & again).

I'd suggest just check the name, `return client.GetName() == onboardingTicketPublicKeySecretName`